source/Octostache/Templates/Functions/TextManipulationFunction.cs

```diff
@@ -137,6 +138,96 @@ internal class TextManipulationFunction
             }
         }
 
+        public static string? Indent(string? argument, string[] options)
+        {
+            if (argument == null)
+            {
+                return null;
+            }
+
+            if (argument.Length == 0)
+            {
+                // No content, no indenting
+                return string.Empty;
+            }
+
+            var indentOptions = new IndentOptions(options);
+
+            if (!indentOptions.IsValid)
+            {
+                return null;
+            }
+
+            return indentOptions.InitialIndent + argument.Replace("\n", "\n" + indentOptions.SubsequentIndent);
```

Comment:
I think we'll need to catch Mac line breaks here, which are just `\r`.

Reply:
Mac last used `\r` as line breaks in OS 9. In current systems, the only line endings to worry about are `\n` and `\r\n` which this solution covers.

Anything beyond that and we're probably also looking at an encoding that isn't supported.

https://en.wikipedia.org/wiki/Newline#Representations

Reply:
👍